pageserver/src/http/routes.rs

```diff
@@ -507,6 +509,36 @@ async fn get_lsn_by_timestamp_handler(
     json_response(StatusCode::OK, result)
 }
 
+async fn get_time_range_of_lsn_handler(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&request, "tenant_id")?;
+    check_permission(&request, Some(tenant_id))?;
+
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+
+    let lsn_str = must_get_query_param(&request, "lsn")?;
+    let lsn = Lsn::from_str(&lsn_str)
+        .with_context(|| format!("Invalid LSN: {lsn_str:?}"))
+        .map_err(ApiError::BadRequest)?;
+
+    let ctx = RequestContext::new(TaskKind::MgmtRequest, DownloadBehavior::Download);
+    let timeline = active_timeline_of_active_tenant(tenant_id, timeline_id).await?;
+    let result = timeline.get_timestamp_range_for_lsn(lsn, &ctx).await?;
+
+    let result = match result {
+        Some((min, max, median)) => {
+            let min = format_rfc3339(postgres_ffi::from_pg_timestamp(min));
+            let max = format_rfc3339(postgres_ffi::from_pg_timestamp(max));
+            let median = format_rfc3339(postgres_ffi::from_pg_timestamp(median));
+            serde_json::json!({ "min": format!("{min}"), "max": format!("{max}"), "median": format!("{median}") })
+        }
+        None => serde_json::json!({ "empty": true }),
```

Comment:
Feels like this could just be json null, or does `empty: false` have some meaning?

Reply:
i've asked on slack and the others suggested using 404 instead. I've adopted that one instead.